sklearn/ensemble/_hist_gradient_boosting/binning.py

```diff
@@ -118,20 +149,29 @@ def fit(self, X, y=None):
         -------
         self : object
         """
-        X = check_array(X, dtype=[X_DTYPE])
+        if not (3 <= self.n_bins <= 256):
```

Comment:
Maybe add a comment like: `3: two distinct bins and a missing values bin`, just to avoid confusion in the future.